File: MVC_O_DiceGame/DiceGame_csharp/DiceGame/controller/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DiceGame.controller
{
	class Player : model.DiceGameObserver
	{
		view.Console m_view;

		public Player(view.Console a_view)
		{
			m_view = a_view;
		}

		public void PlayGame(model.DiceGame a_game)
		{
			a_game.AddSubscriber(this);
			m_view.DisplayInstructions();

			while (m_view.WantsToPlay())
			{
				m_view.DisplayInstructions();

				m_view.DisplayResult(a_game.Play());
				
			}

		}

		public void DiceRolled(int a_faceValue)
		{
			m_view.DisplayDiceValue(a_faceValue);
		}
	}
}

File: MVC_O_DiceGame/DiceGame_java/controller/Player.java
package controller;

public class Player implements model.IDiceRolledObserver {
  
  private view.IView m_view;
  
  public Player(view.IView a_view) {
    m_view = a_view;
  }

  public void playGame(model.DiceGame a_game) {
    
    a_game.addSubscriber(this);
    
    m_view.DisplayWelcomeMessage();
    m_view.DisplayInstructions();
    
    while (m_view.wantsToPlay()) {
      
      m_view.DisplayResult(a_game.play());
      
      m_view.DisplayInstructions();
    }
  }
  
  public void DiceRolled(int a_faceValue) {
    m_view.DisplayDice(a_faceValue);
  }
  
}